cmd/remote/default-package.go

```diff
@@ -53,96 +50,82 @@ func (pkg *defaultPackage) Commands() []command.Command {
 	return pkg.Manifest.Commands()
 }
 
-func (pkg *defaultPackage) InstallTo(targetDir string) (command.PackageManifest, error) {
-	zipReader, _ := zip.OpenReader(pkg.ZipFile)
-	defer zipReader.Close()
-	for _, file := range zipReader.Reader.File {
-		zippedFile, err := file.Open()
-		if err != nil {
-			return nil, fmt.Errorf("installation failed: %s", err)
-		}
-		defer zippedFile.Close()
-
-		extractedFilePath := filepath.Join(targetDir, file.Name)
-		if file.FileInfo().IsDir() {
-			log.Println("Directory Created:", extractedFilePath)
-			err := os.MkdirAll(extractedFilePath, file.Mode())
-			if err != nil {
-				return nil, fmt.Errorf("directory extraction failed: %s", err)
-			}
-		} else {
-			log.Println("File extracted:", file.Name)
-			outputFile, err := os.OpenFile(
-				extractedFilePath,
-				os.O_WRONLY|os.O_CREATE|os.O_TRUNC,
-				file.Mode(),
-			)
-			if err != nil {
-				return nil, fmt.Errorf("file extraction failed: %s", err)
-			}
-			defer outputFile.Close()
-
-			_, err = io.Copy(outputFile, zippedFile)
-			if err != nil {
-				return nil, fmt.Errorf("file data extraction failed: %s", err)
-			}
-		}
+func ReadManifest(file fs.File) (command.PackageManifest, error) {
+	stat, err := file.Stat()
+	if err != nil {
+		return nil, fmt.Errorf("cannot read the manifest file handle (%s)", err)
 	}
 
-	manifestfile, _ := os.Open(filepath.Join(targetDir, "manifest.mf"))
-	defer manifestfile.Close()
+	var payload = make([]byte, stat.Size())
+	nb, err := file.Read(payload)
+	if err != nil && err != io.EOF || nb != int(stat.Size()) {
+		return nil, fmt.Errorf("cannot read the manifest (%s)", err)
+	}
 
-	mf, err := ReadManifest(manifestfile)
+	var mf = defaultPackageManifest{}
+	// YAML is super set of json, should work with JSON as well
+	err = yaml.Unmarshal(payload, &mf)
 	if err != nil {
-		return nil, fmt.Errorf("cannot read package manifest: %s", err)
+		return nil, fmt.Errorf("cannot read the manifest content, it is neither a valid JSON nor YAML (%s)", err)
 	}
 
-	return mf, nil
+	return &mf, nil
 }
 
-func CreatePackage(zipFilename string) (command.Package, error) {
-	reader, err := zip.OpenReader(zipFilename)
+func copyFolder(srcFolder string, dstFolder string) error {
```

Comment:
I would avoid the copy folder operation if possible, see my another comment in git package.